node/test/chan_drain.js

```diff
@@ -739,11 +734,24 @@ function checkFinish(assert, err, cluster, finishCount) {
                 {peers: [{connections: []}]}
             ]
         });
-    }
 
-    if (finishCount <= 0) {
         assert.end();
     }
 
     return finishCount;
 }
+
+function waitForConnRemoved(count, cluster, callback) {
```

Comment:
is there any chance we could just compute the count rather than statically set it? maybe something like:
```
count = 1
  for each chan
    for each peer:
      count += peer.connection.length
      peer.on(..., removed)
removed() // this is to match the "1 to start" above, and ensures that we won't erroneously fire the callback while still iterating chan/peers
```